test/plugin/test_out_secondary_file.rb

```diff
@@ -0,0 +1,393 @@
+require_relative '../helper'
+require 'time'
+require 'fileutils'
+require 'fluent/event'
+require 'fluent/unique_id'
+require 'fluent/plugin/buffer'
+require 'fluent/plugin/out_secondary_file'
+require 'fluent/plugin/buffer/memory_chunk'
+require 'fluent/test/driver/output'
+
+class FileOutputSecondaryTest < Test::Unit::TestCase
+  include Fluent::UniqueId::Mixin
+
+  def setup
+    Fluent::Test.setup
+    FileUtils.rm_rf(TMP_DIR)
+    FileUtils.mkdir_p(TMP_DIR)
+  end
+
+  TMP_DIR = File.expand_path(File.dirname(__FILE__) + "/../tmp/out_secondary_file#{ENV['TEST_ENV_NUMBER']}")
+
+  CONFIG = %[
+    directory #{TMP_DIR}
+    basename out_file_test
+    compress gzip
+  ]
+
+  class DummyOutput < Fluent::Plugin::Output
+    def write(chunk); end
+  end
+
+  def create_primary(buffer_cofig = config_element('buffer'))
+    DummyOutput.new.configure(config_element('ROOT','',{}, [buffer_cofig]))
+  end
+
+  def create_driver(conf = CONFIG, primary = create_primary)
+    c = Fluent::Test::Driver::Output.new(Fluent::Plugin::SecondaryFileOutput)
+    c.instance.acts_as_secondary(primary)
+    c.configure(conf)
+  end
+
+  sub_test_case 'configture' do
+    test 'default configuration' do
+      d = create_driver %[directory #{TMP_DIR}]
+      assert_equal 'dump.bin', d.instance.basename
+      assert_equal TMP_DIR, d.instance.directory
+      assert_equal :text, d.instance.compress
+    end
+
+    test 'should be configurable' do
+      d = create_driver
+      assert_equal 'out_file_test', d.instance.basename
+      assert_equal TMP_DIR, d.instance.directory
+      assert_equal :gzip, d.instance.compress
+    end
+
+    test 'should only use in secondary' do
+      c = Fluent::Test::Driver::Output.new(Fluent::Plugin::SecondaryFileOutput)
+      assert_raise do
+        c.configure(conf)
+      end
+    end
+
+    test 'directory should be writable' do
+      assert_nothing_raised do
+        create_driver %[directory #{TMP_DIR}/test_dir/foo/bar/]
+      end
+
+      assert_nothing_raised do
+        FileUtils.mkdir_p("#{TMP_DIR}/test_dir")
+        File.chmod(0777, "#{TMP_DIR}/test_dir")
+        create_driver %[directory #{TMP_DIR}/test_dir/foo/bar/]
+      end
+
+      assert_raise Fluent::ConfigError do
```

Comment:
`Test::Unit::Assertions#assert_raise_message` can assert erorr message.
But, it can't assert both error class and error message.
I leave it as it is.